### PR TITLE
Subscribe to file path if the editor is gone

### PR DESCRIPTION
**Pull request by 50Wliu**
Even if the editor that the preview tab was tracking has been destroyed, we can still display the rendered markdown by checking its file contents.  Do that instead of instantly destroying the newly-created pane when attempting to split an orphaned preview pane.

While this PR substantially increases the capabilities of Markdown Preview when previewing a file no longer open in Atom, it does not do the following:
* Track when a file is deleted and update the preview accordingly to show an error message
* Go back to using `editorId` if the original file is re-opened in Atom
* (and probably more, though I can't remember anything else right now)

Closes atom/atom#8652

### CODE REVIEW

**Comment by 50Wliu** [comment]
This is ready for review.

File: lib/markdown-preview-view.coffee [hunk 1]
```diff
@@ -132,14 +132,21 @@ class MarkdownPreviewView extends ScrollView
   renderMarkdown: ->
     @showLoading() unless @loaded
     @getMarkdownSource().then (source) => @renderMarkdownText(source) if source?
+    .catch (reason) =>
```
Comment (50Wliu) [inline lib/markdown-preview-view.coffee]: Without this catch the preview pane would endlessly display as loading despite the promise already having been rejected (eg when the file has been deleted).

File: lib/markdown-preview-view.coffee [hunk 2]
```diff
@@ -132,14 +132,21 @@ class MarkdownPreviewView extends ScrollView
...
+        if source is null
+          Promise.reject("#{@file.getBaseName()} could not be found")
+        else
+          Promise.resolve(source)
+      .catch (reason) -> Promise.reject(reason)
     else if @editor?
       Promise.resolve(@editor.getText())
     else
-      Promise.resolve(null)
+      Promise.reject()
```
Comment (50Wliu) [inline lib/markdown-preview-view.coffee]: I don't see any point in resolving a null source.  Better reject the promise to be safe here.

**Comment by 50Wliu** [comment]
The `@file.onDidRename` method seems a bit flaky.  The file was definitely renamed (as seen in the second failing spec), but it looks like the `did-change-title` event was not fired.  In addition I'm not sure why `atom.workspace.getActivePaneItem().getPath()` is returning the old path.  Going to assume that the "Pane has been destroyed" error only occurs when this flakiness happens, as I haven't had any troubles yet while testing locally.

**Comment by 50Wliu** [comment]
Ok, specs are finally passing.

**Comment by nathansobo** [comment]
Sorry for missing this @50Wliu. I didn't comb over every square inch but the tests look thorough and everything seems reasonable. I trust you to be around to address any issues that might arise. 🚢

**Pull request merged**